tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -346,6 +390,29 @@ public static void main(String[] args) {
     }
   }
 
+  private static Metadata createMetadataObj(long started, long completed, boolean status) {
+    String jobStatus = status ? "SUCCEEDED" : "FAILED";
+    String url = "http://" + yarnConf.get("yarn.resourcemanager.webapp.address") + "/cluster/app/" + appIdString;
+    String user;
+    try {
+      user = UserGroupInformation.getCurrentUser().getShortUserName();
+    } catch (IOException e) {
+      LOG.error("User not found. Set user to null", e);
+      user = null;
```

Comment:
Return NULL instead of creating an invalid object?

Reply:
Wdym by invalid object? If the user couldn't be fetched, you would still want the history file to be created, hence I make it null. It just won't show up on the job listing page.

Reply:
By it I mean `user` attribute. The job will still show up, with the user being displayed as `null`.

Reply:
ok, then you can just init `user = null ` and then you don't need to set it here. looks weird.

Reply:
You're right. Fixed: https://github.com/linkedin/TonY/pull/79/commits/3250d31b3f8d01bf283f62e64e92a79a1519dc8e